charts/virtual-kubelet/templates/deployment.yaml

```diff
@@ -30,15 +28,65 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: status.podIP
+{{- if eq .Values.provider "azure" }}
+{{- with .Values.providers.azure }}
+{{- if .targetAKS }}
+        - name: ACS_CREDENTIAL_LOCATION
+          value: /etc/acs/azure.json
+        - name: AZURE_TENANT_ID
+          value: {{ required "tenantId is required" .tenantId }}
+        - name: AZURE_SUBSCRIPTION_ID
+          value: {{ required "subscriptionId is required" .subscriptionId }}
+        - name: AZURE_CLIENT_ID
+          value: {{ required "clientId is required" .clientId }}
+        - name: AZURE_CLIENT_SECRET
+          valueFrom:
+            secretKeyRef:
+              name: {{ template "vk.fullname" $ }}
+              key: clientSecret
+        - name: ACI_RESOURCE_GROUP
+          value: {{ .aciResourceGroup }}
+        - name: ACI_REGION
+          value: {{ .aciRegion }}
+{{- else }}
+        - name: AZURE_AUTH_LOCATION
+          value: /etc/virtual-kubelet/credentials.json
+        - name: ACI_RESOURCE_GROUP
+          value: {{ required "aciResourceGroup is required" .aciResourceGroup }}
+        - name: ACI_REGION
+          value: {{ required "aciRegion is required" .aciRegion }}
+{{- end }}
+{{- end }}
+{{- end }}
         volumeMounts:
         - name: credentials
           mountPath: "/etc/virtual-kubelet"
+{{- if eq .Values.provider "azure" }}
+{{- if .Values.providers.azure.targetAKS }}
+        - name: acs-credential
+          mountPath: "/etc/acs/azure.json"
+{{- end }}
+{{- end }}
         command: ["virtual-kubelet"]
-        args: ["--provider", "azure", "--namespace", {{ default "" .Values.env.monitoredNamespace | quote }}, "--nodename", {{ default "virtual-kubelet" .Values.env.nodeName | quote }} , "--os", {{ default "Linux" .Values.env.nodeOsType | quote }}, "--taint", {{ default "azure.com/aci" .Values.env.nodeTaint | quote }}]
+        args: [
+          "--provider", "{{ required "provider is required" .Values.provider }}",
+          "--namespace", "{{ .Values.monitoredNamespace }}",
+          "--nodename", "{{ required "nodeName is required" .Values.nodeName }}",
```

Comment:
no need to be required, since we've given the default value "virtual-kubelet" in the values.yaml

Reply:
Marking values as required is still desirable in the case that the user overrides with a falsy value or someone removes the default value in the future. This catches that case at install time.